### PR TITLE
[WIP]: Training APIs

### PR DESCRIPTION
**Pull request by sihrc**
need tests

### CODE REVIEW

**Comment by madisonmay** [comment]
Got tests.  Needs one quick round of code review @anneCarlson

File: lib/indico.rb [hunk 1]
```diff
@@ -183,50 +81,90 @@ def self.image_recognition(image, config = nil)
...
+
+  class Collection
+
+      def initialize(collection)
+        @collection = collection
+      end
+
+      def add_data(data, config = nil)
+
+        batch = data[0].kind_of?(Array)
```
Comment (anneCarlson) [inline lib/indico.rb]: `is_batch` is a more rubyish name.

File: lib/indico.rb [hunk 2]
```diff
@@ -183,50 +81,90 @@ def self.image_recognition(image, config = nil)
...
+      end
+
+      def wait(interval = 1)
+        while info()['status'] != "ready" do
+          sleep(interval)
+        end
+      end
+
+      def info(interval = 1)
+        Indico.collections()[@collection]
```
Comment (anneCarlson) [inline lib/indico.rb]: What is this?

Comment (anneCarlson) [inline lib/indico.rb]: We probably want to error catch if they set `@collection` incorrectly.

**Comment by anneCarlson** [comment]
A few notes, but overall looks good!

File: lib/indico.rb (hunk 2 above)
Comment (madisonmay) [inline lib/indico.rb]: Valid point.

Comment (madisonmay) [inline lib/indico.rb]: Although I feel like the standard typeerror raised should be sufficient to let people know what went wrong?

Comment (madisonmay) [inline lib/indico.rb]: I think it's more proper to catch that error at initialization.

**Pull request merged**